### PR TITLE
add eda modules to the collection

### PR DESCRIPTION
**Pull request by jainnikhil30**
This PR adds 5 modules to the eda collection:

- activation
- credential
- decision_environment
- project
- user

### CODE REVIEW

File: plugins/modules/activation.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    type: str
+    required: true
+  extra_vars:
+    description:
+      - The extra variables for the rulebook activation, default is ''.
+    type: str
+    default: ''
+  restart_policy:
+    description:
+      - The restart policy for the rulebook activation, default is 'always'.
```
Comment (bzwei) [inline plugins/modules/activation.py:52]: default is on-failure

File: plugins/modules/activation.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+def main():
+    argument_spec = dict(
+        name=dict(type="str", required=True),
+        description=dict(required=False),
+        project=dict(type="str", required=True),
+        rulebook=dict(type="str", required=True),
+        extra_vars=dict(type="str", default=""),
+        restart_policy=dict(
+            type="str",
+            default="always",
```
Comment (bzwei) [inline plugins/modules/activation.py:109]: default="on-failure"

File: plugins/modules/activation.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+        activation_fields["rulebook_id"] = rulebook_id
+
+    if decision_environment_id is not None:
+        # this is resolved earlier, so save an API call and don't do it again in the loop above
+        activation_fields["decision_environment_id"] = decision_environment_id
+
+    if restart_policy is not None:
+        activation_fields["restart_policy"] = restart_policy
+
+    if restart_policy is not None:
```
Comment (bzwei) [inline plugins/modules/activation.py:185]: duplicated code from line 182

File: plugins/modules/activation.py [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+ANSIBLE_METADATA = {
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = """
+---
+module: activation
+author: "Nikhil Jain (@jainnikhil30)"
+short_description: Create, restart or delete rulebook activations in EDA Controller.
```
Comment (bzwei) [inline plugins/modules/activation.py:21]: Other than create and delete, no other operation such as restart, stop, disable, enable an existing activation is supported in this PR.

File: plugins/modules/activation.py [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+
+- name: Delete EDA Activation
+  ansible.eda.activation:
+    name: "Example Activation"
+    description: "Example Activation description"
+    project: "Example Project"
+    rulebook: "hello_controller.yml"
+    decision_environment: "Default Decision Environment"
+    enabled: False
+    state: absent
```
Comment (bzwei) [inline plugins/modules/activation.py:93]: For absent only name field is required, why do you need to add all other fields in the example? The same for other modules such as credential, decision_environment, project, and user.

File: plugins/modules/activation.py (hunk 1 above)
Comment (jainnikhil30) [reply plugins/modules/activation.py:50]: Sure, made the change

File: plugins/modules/activation.py (hunk 2 above)
Comment (jainnikhil30) [reply plugins/modules/activation.py]: Sure, made the change

File: plugins/modules/activation.py (hunk 3 above)
Comment (jainnikhil30) [reply plugins/modules/activation.py]: My bad, removed it.

File: plugins/modules/activation.py (hunk 4 above)
Comment (jainnikhil30) [reply plugins/modules/activation.py]: That is correct changed the description

File: plugins/modules/activation.py (hunk 5 above)
Comment (jainnikhil30) [reply plugins/modules/activation.py:85]: Yeah the reason I had done it that way, I kept those other fields as required fields. Now I have removed it and you can delete all objects with just name and state as "absent"

**Comment by bzwei** [comment]
LGTM. @jainnikhil30 Please fix the length violations. Is it possible to add tests for the modules?

**Comment by bzwei** [comment]
@jainnikhil30 please resubmit the PR to branch api-module. We're not ready to merge it to main yet. Thanks.

**Comment by jainnikhil30** [comment]
@bzwei Okay, does that mean the modules wont be part of the actual collection release ? If yes, then how would users use the modules ?

**Comment by jainnikhil30** [comment]
Closed this in favor of #178

**Pull request closed without merging**